pkg/controller/cassandradatacenter/cassandradatacenter_controller.go

```diff
@@ -162,6 +162,13 @@ func (r *ReconcileCassandraDataCenter) Reconcile(request reconcile.Request) (rec
 		return reconcile.Result{}, err
 	}
 
+	if rctx.cdc.Spec.PrometheusSupport {
+		err := createOrUpdatePrometheusServiceMonitor(rctx)
```

Comment:
could not you do `if err := _that_method; err != nil { return reconcile.Result{}, err` ?

Reply:
I did it the same way we run other `createOrUpdate...`, but sure, why not.